### PR TITLE
Supports numeric arrays, improves support of domain types and int and varchar array types

### PR DESCRIPTION
**Pull request by Tigrov**
- supports `numeric` arrays
- improves support of domain types
- corrects `precision` and `scale` for arrays of `int`
- corrects `size` for arrays of `varchar` and `char`

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | ❌

### CODE REVIEW

File: tests/SchemaTest.php [hunk 1]
```diff
@@ -525,4 +525,37 @@ public function testNotConnectionPDO(): void
...
+        if ($schema->getTableSchema('test_domain_type') !== null) {
+            $command->dropTable('test_domain_type')->execute();
+        }
+
+        $command->setSql('DROP DOMAIN IF EXISTS sex_char')->execute();
+        $command->setSql(
+            'CREATE DOMAIN sex_char AS "char" NOT NULL DEFAULT \'x\' CHECK (VALUE in (\'m\', \'f\', \'x\'))'
+        )->execute();
+
+        $command->createTable('test_domain_type', ['id' => 'pk', 'sex' => 'sex_char'])->execute();
```
Comment (vjik) [inline tests/SchemaTest.php:545]:
```suggestion
        $command->setSql('DROP DOMAIN IF EXISTS sex_char')->execute();
        $command->setSql(
            'CREATE DOMAIN sex_char AS "char" NOT NULL DEFAULT \'x\' CHECK (VALUE in (\'m\', \'f\', \'x\'))'
        )->execute();
        
        if ($schema->getTableSchema('test_domain_type') !== null) {
            $command->dropTable('test_domain_type')->execute();
        }
        $command->createTable('test_domain_type', ['id' => 'pk', 'sex' => 'sex_char'])->execute();
```

Suggestion only for better reading of test.

**Review by vjik — COMMENTED** [review]
@Tigrov Can you add lines to changelog?

File: src/Schema.php [hunk 2]
```diff
@@ -692,10 +692,10 @@ protected function findColumns(TableSchemaInterface $table): bool
...
             a.attlen AS character_maximum_length,
             pg_catalog.col_description(c.oid, a.attnum) AS column_comment,
-            a.atttypmod AS modifier,
-            a.attnotnull = false AS is_nullable,
-            CAST(pg_get_expr(ad.adbin, ad.adrelid) AS varchar) AS column_default,
-            coalesce(pg_get_expr(ad.adbin, ad.adrelid) ~ 'nextval',false) $orIdentity AS is_autoinc,
+            information_schema._pg_truetypmod(a, t) AS modifier,
+            NOT (a.attnotnull OR t.typnotnull) AS is_nullable,
+            COALESCE(t.typdefault, pg_get_expr(ad.adbin, ad.adrelid)) AS column_default,
+            COALESCE(pg_get_expr(ad.adbin, ad.adrelid) ~ 'nextval', false) $orIdentity AS is_autoinc,
```
Comment (darkdef) [inline src/Schema.php:698]: Good solution. It's adopt from SQL to internal functions?
As https://gemfury.com/schabrolles/rpm:postgresql96-server/-/content/usr/pgsql-9.6/share/information_schema.sql

File: tests/ColumnSchemaTest.php [hunk 3]
```diff
@@ -65,6 +66,7 @@ public function testPhpTypeCast(): void
         $boolColPhpTypeCast = $tableSchema->getColumn('bool_col')?->phpTypecast($query['bool_col']);
         $numericColPhpTypeCast = $tableSchema->getColumn('numeric_col')?->phpTypecast($query['numeric_col']);
         $intArrayColPhpType = $tableSchema->getColumn('intarray_col')?->phpTypecast($query['intarray_col']);
+        $numericArrayColPhpTypeCast = $tableSchema->getColumn('numericarray_col')?->phpTypecast($query['numericarray_col']);
```
Comment (darkdef) [inline tests/ColumnSchemaTest.php:69]: Please add more test cases
for varchar array also

**Review by darkdef — CHANGES_REQUESTED** [review]
And please - fix statical analyzer problems

File: src/Schema.php (hunk 2 above)
Comment (Tigrov) [reply src/Schema.php:698]: Yes, the functions `_pg_numeric_precision` and `_pg_numeric_scale` have the same code and it looks more accurate to use them.
https://github.com/postgres/postgres/blob/master/src/backend/catalog/information_schema.sql

File: tests/ColumnSchemaTest.php (hunk 3 above)
Comment (Tigrov) [reply tests/ColumnSchemaTest.php:70]: Ok, ready

**Review by darkdef — APPROVED** [review]
Good job

**Review by terabytesoftw — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,8 +2,8 @@
 
 ## 1.0.1 under development
 
-- no changes in this release.
+- Enh #282: Supports `numeric` arrays, improves support of domain types and `int` and `varchar` array types (Tigrov)
```
Comment (vjik) [inline CHANGELOG.md:5]:
```suggestion
- Enh #282: Supports `numeric` arrays, improves support of domain types and `int` and `varchar` array types (@Tigrov)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,8 +2,8 @@
 
 ## 1.0.1 under development
 
-- no changes in this release.
+- Enh #282: Supports `numeric` arrays, improves support of domain types and `int` and `varchar` array types (@Tigrov)
```
Comment (samdark) [inline CHANGELOG.md:5]:
```suggestion
- Enh #282: Support `numeric` arrays, improve support of domain types and `int` and `varchar` array types (@Tigrov)
```

**Pull request merged**